c7n/actions/webhook.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2019 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+
+import jmespath
+from six.moves.urllib import request, parse
+from six.moves.urllib.error import URLError
+
+from c7n import utils
+from .core import EventAction
+
+
+class Webhook(EventAction):
+    """Calls a webhook with optional parameters and body
+       populated from JMESPath queries.
+
+        .. code-block:: yaml
+
+          policies:
+            - name: call-webhook
+              resource: ec2
+              description: |
+                Call webhook with list of resource groups
+              actions:
+               - type: webhook
+                 url: http://foo.com
+                 parameters:
+                    - resource_name: name
+    """
+
+    schema = utils.type_schema(
+        'webhook',
+        required=['url'],
+        **{
+            'url': {'type': 'string'},
+            'body': {'type': 'string'},
+            'batch': {'type': 'boolean'},
+            'method': {'type': 'string'},
+            'parameters': {
+                "type": "object",
+                "additionalProperties": {
+                    "type": "string",
+                    "description": "query string values"
+                }
+            }
+        }
+    )
+
+    def get_permissions(self):
+        return ()
+
+    def process(self, resources, event=None):
+        self.url = self.data['url']
+        self.body = self.data.get('body')
+        self.batch = self.data.get('batch', False)
+
+        if self.batch:
+            self._process_call(resources)
+        else:
+            for r in resources:
+                self._process_call(r)
+
+    def _process_call(self, resource):
+        prepared_url = self._build_url(resource)
+        prepared_body = self._build_body(resource)
+        method = self.data.get('method', 'POST' if prepared_body else 'GET')
```

Comment:
hmm so in a get, what are we passing? we're just invoking a url without information?